src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts

```diff
@@ -104,24 +107,53 @@ export function useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAnd
       ),
     []
   )
-  const { loading, error, data } = useQuery<QueryResult>(QUERY, {
-    variables: {
-      lowerTimeLimit: memoizedLowerTimeLimit,
-      userId: account?.toLowerCase() || ''
-    }
-  })
+  const subgraphAccountId = useMemo(() => account?.toLowerCase() || '', [account])
   const filterTokenAddress = useMemo(() => tokenFilter?.address.toLowerCase(), [tokenFilter])
 
+  const [loadingPairs, setLoadingPairs] = useState(false)
+  const [pairs, setPairs] = useState<SubgraphPair[]>([])
+
+  useEffect(() => {
+    let cancelled = false
+    const fetchData = async () => {
+      if (!chainId) return
+      const pairs = []
+      let lastId = ''
+      setLoadingPairs(true)
+      setPairs([])
+      try {
+        while (1) {
```

Comment:
Wondering whats the reason for this while loop?

Reply:
It's basically to make sure to fetch ALL the pairs. It breaks as soon as we encounter a non-full page. Until then, new paginated requests are made to fetch everything.